src/sp/transport/mqttws/nmq_websocket.c

```diff
@@ -353,6 +353,15 @@ wstran_pipe_recv_cb(void *arg)
 	nni_aio_finish(uaio, 0, nni_msg_len(smsg));
 	return;
 reset:
+	// If the connection is closed, we need to pass back a different
+	// error code.  This is necessary to avoid a problem where the
+	// closed status is confused with the accept file descriptor
+	// being closed.
+	// When Listener's accept_cb received NNG_ECLOSED, listerner will be closed...
+	if (rv == NNG_ECLOSED) {
+		rv = SERVER_SHUTTING_DOWN;
+	}
+
```

Comment:
A question. Is it reasonable to pass SERVER_SHUTTING_DOWN rv to p->ep_aio?

Reply:
The error code doesn't have to be SERVER_SHUTTING_DOWN.
It just needs to be able to continue executing listerner start when the listener receives it.